pkg/cluster/internal/providers/podman/provider.go

```diff
@@ -430,8 +430,10 @@ func info(logger log.Logger) (*providers.ProviderInfo, error) {
 		SupportsCPUShares:   cgroupSupportsCPUShares,
 	}
 	if info.Rootless && !v.AtLeast(version.MustParseSemantic("4.0.0")) {
-		logger.Warn("Cgroup controller detection is not implemented for Podman. " +
-			"If you see cgroup-related errors, you might need to set systemd property \"Delegate=yes\", see https://kind.sigs.k8s.io/docs/user/rootless/")
+		if logger != nil {
```

Comment:
the logger is only used, here, since we need to use the `info(logger log.Logger)` function to detect if it is rootless, better to make the logging optional, when podman < 4.0 is deprecated we don't need this logger anymore

Reply:
logger is a standard part of the provider /shrug